### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by maarcingebala**
Update the changelog with recent releases. Also changed obsolete `3.1.0-a.0` to `dev` that was hardcoded as the version number on `main`.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,34 +2,84 @@
 
 All notable, unreleased changes to this project will be documented in this file. For the released changes, please visit the [Releases](https://github.com/mirumee/saleor/releases) page.
 
-
 # Unreleased
 
+# 3.2.0
+
 ### Breaking changes
-- PREVIEW_FEATURE: replace error code `NOT_FOUND` with `CHECKOUT_NOT_FOUND` for mutation `OrderCreateFromCheckout` - #9569 by @korycins
```
Comment (korycins) [inline CHANGELOG.md:9]: Shouldn't we move this to `unreleased`?

Comment (maarcingebala) [reply CHANGELOG.md:9]: Fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,34 +2,93 @@
...
+- Add subscription webhooks (#9394) @jakubkuc
+- Add `language_code` field to webhook payload for `Order`, `Checkout` and `Customer` - #9433 by @rafalp
 - Refactor app tokens - #9438 by @IKarbowiak
   - Store app tokens hashes instead of plain text.
-- Fix filtering product attributes by date range - #9543 by @IKarbowiak
-- Save images to product media from external URLs - #9329 by @krzysztofwolski
-- Fix for raising Permission Denied when anonymous user calls `checkout.customer` field - #9573 by @korycins
+- Add category webhook events - #9490 by @SzymJ
+- Fix access to own resources by App - #9425 by @korycins
+- Add `handle_checkouts` permission - #9402 by @SzymJ
```
Comment (SzymJ) [inline CHANGELOG.md:41]: I think this one was done by @[korycins](https://github.com/korycins)

Comment (maarcingebala) [reply CHANGELOG.md]: Thanks, fixed.

**Review by IKarbowiak — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,34 +2,93 @@
...
+- Port a better schema printer from GraphQL Core 3.x - #9389 by @patrys
+- Fix failing `checkoutCustomerAttach` mutation - #9401 by @IKarbowiak
+- Add new mutation `orderCreateFromCheckout` - #9343 by @korycins
+- Assign missing user to context - #9520 by @korycins
+- Add default ordering to order discounts - #9517 by @fowczarek
+- Raise formatted error when trying to assign assigned media to variant - #9496 by @L3str4nge
+- Update `orderNumber` field in `OrderEvent` type - #9447 by @IKarbowiak
+- Do not create `AttributeValues` when values are not provided - #9446 @IKarbowiak
+- Add response status code to event delivery attempt - #9456 by @przlada
+- Don't rely on counting counting objects when reindexing - #9442 by @patrys
```
[the remark this answers is not in the document]
Comment (maarcingebala) [reply CHANGELOG.md]: Fixed

**Pull request merged**